File: src/prod/src/client/StartNodeResult.cpp
// ------------------------------------------------------------
// Copyright (c) Microsoft Corporation.  All rights reserved.
// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
// ------------------------------------------------------------

#include "stdafx.h"

using namespace std;
using namespace Common;
using namespace Api;
using namespace ServiceModel;
using namespace Management::FaultAnalysisService;
using namespace Client;

StartNodeResult::StartNodeResult(
    StartNodeStatus && nodeStatus)
    : nodeStatus_(move(nodeStatus))
{
}

shared_ptr<NodeResult> const & StartNodeResult::GetNodeResult()
{
    return nodeStatus_.GetNodeResult();
}

File: src/prod/src/managed/Api/src/System/Fabric/Result/StartNodeResult.cs
// ------------------------------------------------------------
// Copyright (c) Microsoft Corporation.  All rights reserved.
// Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
// ------------------------------------------------------------

namespace System.Fabric.Result
{
    using System.Numerics;

    /// <summary>
    /// Returns Start node result object.
    /// </summary>
    /// <remarks>
    /// This class returns the NodeResult for Start Node action.
    /// </remarks>
    [Serializable]
    public class StartNodeResult : NodeResult
    {
        /// <summary>
        /// Start node result constructor.
        /// </summary>
        /// <param name="nodeName">Node name</param>
        /// <param name="nodeInstance">node instance id</param>
        internal StartNodeResult(string nodeName, BigInteger nodeInstance)
            : base(nodeName, nodeInstance)
        {
        }
    }
}